eclass/cargo.eclass

```diff
@@ -129,7 +125,14 @@ cargo_live_src_unpack() {
 
 # @FUNCTION: cargo_gen_config
 # @DESCRIPTION:
-# Generate the $CARGO_HOME/config necessary to use our local registry
+# Generate the $CARGO_HOME/config necessary to use our local registry and settings.
+# Cargo can also be configured through environment variables in addition to the TOML syntax below.
+# For each configuration key below of the form foo.bar the environment variable CARGO_FOO_BAR
+# can also be used to define the value.
+# Environment variables will take precedent over TOML configuration,
+# and currently only integer, boolean, and string keys are supported.
+# For example the build.jobs key can also be defined by CARGO_BUILD_JOBS.
+# Or setting CARGO_TERM_VERBOSE=false in make.conf will make build quieter.
```

Comment:
Should we doc these officially with:
```
@ECLASS-VARIABLE: CARGO_BUILD_JOBS
@DEFAULT_UNSET
@DESCRIPTION
Overrides the default MAKEOPTS -j value

@ECLASS-VARIABLE: CARGO_TERM_VERBOSE
@DEFAULT_UNSET
@DESCRIPTION
Set this to false to make the build process quieter.
```

Reply:
No IMO, those are strictly cargo variables, we just now have eclass that does not override them via commandline. I adopted upstream documentation pieces I think it should be ok as is, and it even will make it into `eclass-manpages` automatically.

I think we are a bit behind on documentation on wiki, I'll document user settings there, wiki is discoverable for users.